### PR TITLE
Cov 102

### PR DESCRIPTION
**Pull request by andrewbird**
I think the ne2000 is a real bug.

### CODE REVIEW

File: src/base/core/ports.c [hunk 1]
```diff
@@ -390,6 +390,7 @@ Bit8u std_port_inb(ioport_t port)
 	}
 	pr.port = port;
 	pr.type = TYPE_INB;
+	pr.word = 0;
```
Comment (stsp) [inline src/base/core/ports.c]: Its better to just init `pr` with `= {}`.

File: src/base/lib/misc/dlmalloc.c [hunk 2]
```diff
@@ -3366,7 +3366,7 @@ void* dlmemalign(size_t alignment, size_t bytes) {
 
 void** dlindependent_calloc(size_t n_elements, size_t elem_size,
                                  void* chunks[]) {
-  size_t sz = elem_size; /* serves as 1-element array */
+  size_t sz[1] = {elem_size,}; /* 1-element array */
   return ialloc(gm, n_elements, &sz, 3, chunks);
```
Comment (stsp) [inline src/base/lib/misc/dlmalloc.c]: In that case you don't need `&`.

Comment (andrewbird) [reply src/base/lib/misc/dlmalloc.c]: okay, done

File: src/base/core/ports.c (hunk 1 above)
Comment (andrewbird) [reply src/base/core/ports.c]: okay, done

File: src/base/dev/ne2k/ne2000.c [hunk 3]
```diff
@@ -670,7 +670,7 @@ static inline void ne2000_mem_writew(NE2000State *s, uint32_t addr,
 
     addr &= ~1; /* XXX: check exact behaviour if not even */
     if (addr < 32 ||
-        (addr >= NE2000_PMEM_START && addr < NE2000_MEM_SIZE)) {
+        (addr >= NE2000_PMEM_START && addr < (NE2000_MEM_SIZE - 2))) {
```
Comment (stsp) [inline src/base/dev/ne2k/ne2000.c:673]: I believe these all should be `<=`.

Comment (andrewbird) [reply src/base/dev/ne2k/ne2000.c:673]: Interesting, I just checked the latest qemu code and they seem to have one of each
https://github.com/qemu/qemu/blob/23901b2b721c0576007ab7580da8aa855d6042a9/hw/net/ne2000.c#L473-L494

Comment (stsp) [reply src/base/dev/ne2k/ne2000.c:673]: They do it correctly.

**Pull request closed without merging**